### ISSUE TITLE
[7plus] Olympic Replays not playing

### ISSUE THREAD

**Issue by Arathen** [matthuisman/slyguy.addons#39]
Many thanks for these Addons. Ive been a long time fan of your Kayo addon particularly. Ive got a problem with the 7plus addon since the Olympics started though.

"Live Olympics" work fine. No issues there. Likewise, "Latest From Tokyo" and "Highlights" all work fine as well. Watching replays on the 7 website in a browser also works fine. But when you navigate down into "Full Replays" or "Tokyo Minis" in the addon, and try to play any stream under those sections, the following error is all you get.

```
2021-07-28 18:26:46.564 T:3365695712   ERROR: slyguy.7plus - 'NoneType' object is not iterable
                                            Traceback (most recent call last):
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/signals.py", line 31, in throwable
                                                yield
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/router.py", line 113, in dispatch
                                                function(**params)
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/plugin.py", line 65, in decorated_function
                                                item = f(*args, **kwargs)
                                              File "/home/osmc/.kodi/addons/slyguy.7plus/resources/lib/plugin.py", line 315, in play_vod
                                                for row2 in row.get('items', []):
                                            TypeError: 'NoneType' object is not iterable
2021-07-28 18:26:50.724 T:4072886288  NOTICE: VideoPlayer::OpenFile: plugin://slyguy.7plus/?_=play_vod&_play=1&slug=olympic-games-tokyo-swimming%3Fepisode-id%3D7SWM01-010
2021-07-28 18:26:50.725 T:3344711904  NOTICE: Creating InputStream
2021-07-28 18:26:50.726 T:3344711904   ERROR: Open - failed to open source <http>
2021-07-28 18:26:50.726 T:3344711904   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://slyguy.7plus/?_=play_vod&_play=1&slug=olympic-games-tokyo-swimming%3Fepisode-id%3D7SWM01-010]
2021-07-28 18:26:50.727 T:3344711904  NOTICE: CVideoPlayer::OnExit()
2021-07-28 18:26:50.840 T:4072886288  NOTICE: CVideoPlayer::CloseFile()
2021-07-28 18:26:50.840 T:4072886288  NOTICE: VideoPlayer: waiting for threads to exit
2021-07-28 18:26:50.840 T:4072886288  NOTICE: VideoPlayer: finished waiting
2021-07-28 18:26:52.842 T:3365695712   ERROR: slyguy.7plus - 'NoneType' object is not iterable
                                            Traceback (most recent call last):
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/signals.py", line 31, in throwable
                                                yield
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/router.py", line 113, in dispatch
                                                function(**params)
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/plugin.py", line 65, in decorated_function
                                                item = f(*args, **kwargs)
                                              File "/home/osmc/.kodi/addons/slyguy.7plus/resources/lib/plugin.py", line 315, in play_vod
                                                for row2 in row.get('items', []):
                                            TypeError: 'NoneType' object is not iterable
2021-07-28 18:26:58.790 T:4072886288  NOTICE: VideoPlayer::OpenFile: plugin://slyguy.7plus/?_=play_vod&_play=1&slug=olympic-games-tokyo-basketball%3Fepisode-id%3D7BKB01-008
2021-07-28 18:26:58.793 T:3635925216  NOTICE: Creating InputStream
2021-07-28 18:26:58.794 T:3635925216   ERROR: Open - failed to open source <http>
2021-07-28 18:26:58.794 T:3635925216   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://slyguy.7plus/?_=play_vod&_play=1&slug=olympic-games-tokyo-basketball%3Fepisode-id%3D7BKB01-008]
2021-07-28 18:26:58.794 T:3635925216  NOTICE: CVideoPlayer::OnExit()
2021-07-28 18:26:58.904 T:4072886288  NOTICE: CVideoPlayer::CloseFile()
2021-07-28 18:26:58.904 T:4072886288  NOTICE: VideoPlayer: waiting for threads to exit
2021-07-28 18:26:58.904 T:4072886288  NOTICE: VideoPlayer: finished waiting
2021-07-28 18:27:37.312 T:3365695712   ERROR: slyguy.7plus - 'NoneType' object is not iterable
                                            Traceback (most recent call last):
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/signals.py", line 31, in throwable
                                                yield
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/router.py", line 113, in dispatch
                                                function(**params)
                                              File "/home/osmc/.kodi/addons/script.module.slyguy/resources/modules/slyguy/plugin.py", line 65, in decorated_function
                                                item = f(*args, **kwargs)
                                              File "/home/osmc/.kodi/addons/slyguy.7plus/resources/lib/plugin.py", line 315, in play_vod
                                                for row2 in row.get('items', []):
                                            TypeError: 'NoneType' object is not iterable
2021-07-28 18:27:40.372 T:4072886288  NOTICE: VideoPlayer::OpenFile: plugin://slyguy.7plus/?_=play_vod&_play=1&slug=olympic-games-tokyo-rowing%3Fepisode-id%3D7ROW01-012
2021-07-28 18:27:40.373 T:3197104352  NOTICE: Creating InputStream
2021-07-28 18:27:40.374 T:3197104352   ERROR: Open - failed to open source <http>
2021-07-28 18:27:40.374 T:3197104352   ERROR: CVideoPlayer::OpenInputStream - error opening [plugin://slyguy.7plus/?_=play_vod&_play=1&slug=olympic-games-tokyo-rowing%3Fepisode-id%3D7ROW01-012]
2021-07-28 18:27:40.374 T:3197104352  NOTICE: CVideoPlayer::OnExit()
2021-07-28 18:27:40.504 T:4072886288  NOTICE: CVideoPlayer::CloseFile()
2021-07-28 18:27:40.520 T:4072886288  NOTICE: VideoPlayer: waiting for threads to exit
2021-07-28 18:27:40.520 T:4072886288  NOTICE: VideoPlayer: finished waiting
```

Any help or advice would be appreciated.

Many thanks!

**Comment by matthuisman** [comment]
fixed in 0.2.0 (https://github.com/matthuisman/slyguy.addons/commit/3aec5a816291964f9373d247f699e5f295f06d41)

How to force update:
https://www.matthuisman.nz/2021/07/force-update-slyguy-add-ons.html

**Comment by Arathen** [comment]
Looks good. Thanks heaps for such a quick fix on this @matthuisman . Its very much appreciated. Cheers.
